float32 pos
bool upper_limit
bool lower_limit
bool origin_point